sql/moz-fx-data-shared-prod/braze_external/changed_waitlists_sync_v1/query.sql

```diff
@@ -21,15 +23,15 @@ SELECT
         )
         ORDER BY
           waitlists_array.update_timestamp DESC
-      ) AS newsletters
+      ) AS waitlists
     )
   ) AS PAYLOAD
 FROM
-  `moz-fx-data-shared-prod.braze_derived.waitlists_v1` AS waitlists,
-  -- CROSS JOIN
-  UNNEST(waitlists.waitlists) AS waitlists_array
-  -- max_update
-  -- WHERE
-  --   waitlists_array.update_timestamp > max_update.max_update_timestamp
+  `moz-fx-data-shared-prod.braze_derived.waitlists_v1` AS waitlists
+CROSS JOIN
+  UNNEST(waitlists.waitlists) AS waitlists_array,
+  max_update
```

Comment:
Why does `max_update` need to be cross joined as well?

Reply:
By cross joining max_update with the unnested waitlists, the max_update_timestamp is made available as a constant column across all the resultant rows

Typically cross joining multiplies the number of rows in the result set, but in this case the max_update CTE is guaranteed to only ever contain one (or none) row. In this instance it isn't increasing the number of rows processed beyond what is already produced by unnesting waitlists

An inner join would work as well, or a subquery in a where clause

Reply:
`max_update_timestamp` isn't used though is it? Since you switched to the subquery

Reply:
Oh, I see! Yes, updating, ty!